spec/services/preserved_object_handler_check_exist_spec.rb

```diff
@@ -394,13 +394,45 @@
         po_handler.check_existence
         expect(Rails.logger).to have_received(:debug).with("check_existence #{druid} called")
       end
+
+      context 'CompleteMoab path does not match moab on disk' do
+        let(:druid) { 'bj102hs9687' }
+        let(:wrong_ms_root) { MoabStorageRoot.find_by!(storage_location: 'spec/fixtures/storage_root02/sdr2objects') }
+        let(:cm) { po.complete_moabs.find_by!(moab_storage_root: ms_root) }
+        let(:po_handler) { described_class.new(druid, incoming_version, incoming_size, wrong_ms_root) }
+
+        before do
+          allow(WorkflowReporter).to receive(:report_completed)
+          cm.moab_storage_root = wrong_ms_root
+          cm.version = 3
+          cm.save!
+        end
+
+        it 'returns expected results' do
+          results = po_handler.check_existence
+          expect(results).to include(a_hash_including(AuditResults::CM_STATUS_CHANGED =>
+                                                          'CompleteMoab status changed from ok to online_moab_not_found'))
+          expect(results).to include(a_hash_including(AuditResults::MOAB_NOT_FOUND))
+          expect(results).to include(a_hash_including(AuditResults::CM_STORAGE_ROOT_CHANGED))
+          expect(results).to include(a_hash_including(AuditResults::CM_STATUS_CHANGED =>
+                                                          'CompleteMoab status changed from online_moab_not_found to ok'))
+        end
+
+        it 'changes the storage root' do
+          po_handler.check_existence
+          cm.reload
+          expect(cm.moab_storage_root).to eq ms_root
+        end
+      end
     end
 
     context 'object not in db' do
-      let(:exp_po_not_exist_msg) { "PreservedObject db object does not exist" }
+      let(:exp_po_not_exist_msg) { "#<ActiveRecord::RecordNotFound: Couldn't find PreservedObject> db object does not exist" }
       let(:exp_obj_created_msg) { "added object to db as it did not exist" }
 
       context 'presume validity and test other common behavior' do
+        let(:druid) { 'bp628nk4868' }
```

Comment:
not sure i understand the need for this `let`... would `druid` have been `ab123cd4567`, but now you need something real from the fixtures?

Reply:
Correct, I need something that matches the fixtures.